File: tests/fixtures/lua/control-flow.lua
local function helper()
end

for index = startValue(), endValue(), stepValue() do
  index()
  local scoped = function()
    helper()
  end
  scoped()
end
index()
scoped()

for key, value in iterate() do
  key()
  value()
end
key()
value()

repeat
  local repeated = function()
    helper()
  end
  repeated()
until repeated()
repeated()

before()
local before = function()
  helper()
end
before()

local callback = function()
  helper()
end
callback()
callback = replacement
callback()

local first, second = helper, helper
first()
second()

File: tests/fixtures/php/control-flow.php
<?php
function control(array $callbacks, callable $replacement): void
{
    foreach ($callbacks as $key => $callback) {
        $key();
        $callback();
    }
    $key();
    $callback();

    foreach ($callbacks as [$left, $right]) {
        $left();
        $right();
    }
    $left();
    $right();

    for ($index = start(); check($index); advance($index)) {
        $index();
    }
    $index();

    try {
        risky();
    } catch (Throwable $failure) {
        $failure();
    }
    $failure();

    if (condition()) {
        $conditional = fn() => build();
        $conditional();
    }
    $conditional();

    $before();
    $before = fn() => build();
    $before();

    $callable = fn() => build();
    $callable();
    $callable = $replacement;
    $callable();
    unset($callable);
    $callable();

    [$first, $second] = $callbacks;
    $first();
    $second();

    global $globalCallback;
    $globalCallback();
    static $staticCallback;
    $staticCallback();
}
